### ISSUE TITLE
Update to latest wagmi/core, some typescript issues to tackle

### ISSUE THREAD

**Issue by Yuripetusko** [TalismanSociety/talisman-connectors#3]
Tried submitting a PR but there's more work required than I hoped for

<img width="969" alt="image" src="https://user-images.githubusercontent.com/666169/198011972-26f5c7af-54fe-48b5-834a-dad3e0a9a464.png">

**Comment by Yuripetusko** [comment]
The reason why I wanted to update wagmi in the first place is that using the Talisman connector right now in our project gives a TS error, as wagmi now expects a template literal types for `account` as `0x${string}` instead of just `string`

<img width="837" alt="image" src="https://user-images.githubusercontent.com/666169/198014288-2350be95-496b-4938-b0d3-c9ebbf133948.png">

**Comment by 0xKheops** [comment]
as temporary workaround, cast as any : 

```ts
export const talismanWallet = ({ chains, shimDisconnect }: MyWalletOptions): Wallet => ({
  id: 'talisman',
  name: 'Talisman',
  iconUrl: imgTalisman,
  iconBackground: '#D5FF5C',
  downloadUrls: {
    browserExtension: 'https://talisman.xyz/download'
  },
  createConnector: () => {
    const connector = new TalismanConnector({
      chains: chains as any, // they are compatible
      options: {
        shimDisconnect
      }
    })

    return {
      connector: connector as any // they are compatible
    }
  }
})
```

sample dapp using rainbowkit & wagmi here : https://github.com/TalismanSociety/talisman-connectors/tree/main/apps/rainbowkit-example

**Comment by 0xKheops** [comment]
Fixed with @talismn/wagmi-connector version 0.1.0
no `as any` required anymore
